app/controlplane/internal/biz/organization.go

```diff
@@ -41,25 +45,96 @@ type OrganizationUseCase struct {
 	logger            *log.Helper
 	casBackendUseCase *CASBackendUseCase
 	integrationUC     *IntegrationUseCase
+	membershipRepo    MembershipRepo
 }
 
-func NewOrganizationUseCase(repo OrganizationRepo, repoUC *CASBackendUseCase, iUC *IntegrationUseCase, logger log.Logger) *OrganizationUseCase {
+func NewOrganizationUseCase(repo OrganizationRepo, repoUC *CASBackendUseCase, iUC *IntegrationUseCase, mRepo MembershipRepo, logger log.Logger) *OrganizationUseCase {
 	return &OrganizationUseCase{orgRepo: repo,
 		logger:            log.NewHelper(logger),
 		casBackendUseCase: repoUC,
 		integrationUC:     iUC,
+		membershipRepo:    mRepo,
 	}
 }
 
 func (uc *OrganizationUseCase) Create(ctx context.Context, name string) (*Organization, error) {
 	// Create a random name if none is provided
 	if name == "" {
 		name = namesgenerator.GetRandomName(0)
+		// Replace underscores with dashes to make it compatible with DNS1123
+		name = strings.ReplaceAll(name, "_", "-")
+	}
+
+	if err := validateOrgName(name); err != nil {
+		return nil, NewErrValidation(fmt.Errorf("invalid organization name: %w", err))
 	}
 
 	return uc.orgRepo.Create(ctx, name)
 }
 
+func validateOrgName(name string) error {
```

Comment:
and the new validation logic performed on creation and update.

Note that selecting the name on creation is not yet supported, but will.

That's why for the auto-generated names we make sure it complies with this new format.